src/gbulb/transports.py

```diff
@@ -387,8 +408,29 @@ def __init__(self, loop, channel, protocol, waiter, extra):
         super().__init__(loop, None, protocol, waiter, extra)
 
     def _create_read_future(self, size):
+        if self._alloc_read_buffers:
+            self._read_buffer = self._protocol.get_buffer(size)
+            size = len(self._read_buffer)
         return self._loop._channel_read(self._channel, size)
 
+    def _submit_read_data(self, data):
+        assert isinstance(data, bytes)
+        if data != b"" and data != 0:
+            if self._alloc_read_buffers:
+                #FIXME: GLib does not actually expose the equivalent to
```

Comment:
Note: This line caused `flake8` to cry with error E265, because there is no space between the hash sign and the word FIXME…